osu.Game/Screens/SelectV2/BeatmapTitleWedge_DifficultyDisplay.cs

```diff
@@ -254,15 +253,51 @@ private void updateDisplay()
                     difficultyText.Text = beatmap.Value.BeatmapInfo.DifficultyName;
                     mapperLink.Action = () => linkHandler?.HandleLink(new LinkDetails(LinkAction.OpenUserProfile, beatmap.Value.Metadata.Author));
                     mapperText.Text = beatmap.Value.Metadata.Author.Username;
+                }
 
-                    var playableBeatmap = beatmap.Value.GetPlayableBeatmap(ruleset.Value);
+                updateStarDifficulty(cancellationSource.Token);
+                updateCountStatistics(cancellationSource.Token);
+                updateDifficultyStatistics();
+            }
+
+            private void updateStarDifficulty(CancellationToken cancellationToken)
+            {
+                difficultyCache.GetDifficultyAsync(beatmap.Value.BeatmapInfo, ruleset.Value, mods.Value, cancellationToken)
+                               .ContinueWith(task =>
+                               {
+                                   Schedule(() =>
+                                   {
+                                       if (cancellationToken.IsCancellationRequested)
+                                           return;
 
-                    countStatisticsDisplay.Statistics = playableBeatmap.GetStatistics()
-                                                                       .Select(s => new StatisticDifficulty.Data(s.Name, s.BarDisplayLength ?? 0, s.BarDisplayLength ?? 0, 1, s.Content))
-                                                                       .ToList();
+                                       starRatingDisplay.Current.Value = task.GetResultSafely() ?? default;
+                                   });
+                               }, cancellationToken);
+            }
+
+            private void updateCountStatistics(CancellationToken cancellationToken)
+            {
+                if (beatmap.IsDefault)
+                {
+                    countStatisticsDisplay.Statistics = Array.Empty<StatisticDifficulty.Data>();
+                    return;
                 }
 
-                updateDifficultyStatistics();
+                Task.Run(() =>
+                {
+                    var playableBeatmap = beatmap.Value.GetPlayableBeatmap(ruleset.Value);
```

Comment:
This causes the playable beatmap to be loaded completely twice. There's gotta be a better way forward than this, but it'll require some thinking. It would be great if `BeatmapDifficultyCache` could return its playable beatmap inside or alongside `StarDifficulty`, but I'm open to other solutions.

Reply:
I don't have any clever idea for this myself. I followed suit with old implementation, which suffers from the same issue of loading beatmap twice.

Reply:
Did I say two? It's actually three..